src/managers.py

```diff
@@ -0,0 +1,69 @@
+from git import Repo
+from joblib import Parallel, delayed
+from tqdm import tqdm
+
+from src.constants import n_jobs, VENDOR_PATH
+from src.models import File
+from src.providers import LanguagesProvider
+from src.services import GitService, GithubService, JsonService
+from src.setup import setup
+
+
+class DeveloperVectorManager:
+    """
+    A class to manage developer vectors.
+    """
+
+    def __init__(
+            self,
+            git_service: GitService,
+            github_service: GithubService,
+    ) -> None:
+        self.git_service = git_service
+        self.github_service = github_service
+
+    def clone_all_user_repos(self, username: str) -> [Repo]:
+        repos = list(self.github_service.get_repos(username))
+        Parallel(n_jobs=n_jobs)(
+            delayed(self.git_service.clone_repo)(repo.clone_url, VENDOR_PATH + 'repos/' + repo.name)
+            for repo in tqdm(repos)
+        )
+        return repos
+
+    def get_repo_files_languages(self, repo_path: str) -> [File]:
+        lp = LanguagesProvider(repo_path)
+        lp.calc_languages()
+        return lp.files_with_languages
+
+    def get_file_authors(self, file: File, repo_path: str) -> File:
```

Comment:
let's use git history in a different way
* traverse git history
* filter out commits where you have more than 1 parent (merge commits)
* find diffs for each commit - and present it in a way like `{author: "some_name", repo_name: "repository name", commit_sha: "commit_sha", filepath: "filepath", added: "number of added lines", deleted: "number of deleted lines"}`